ntoskrnl/ex/handle.c

```diff
@@ -18,6 +18,8 @@
 LIST_ENTRY HandleTableListHead;
 EX_PUSH_LOCK HandleTableListLock;
 #define SizeOfHandle(x) (sizeof(HANDLE) * (x))
+#define MAKE_HANDLE(x) ((x) << 2)
+#define CLEAR_HANDLE(x) MAKE_HANDLE(x.Index)
```

Comment:
Why is this defined like that?

Reply:
Well, you a right, it's a bit unclear. Probably the best way to write this will be
```
#define MAKE_HANDLE(x) ((x) << HANDLE_TAG_BITS)
```
so it will be more obvious. 

BTW, the error with handles was quite interesting: it's one of those "32-bit chauvinism" cases, there the code was working just by chance of `SizeOfHandle(1) == (1 << HANDLE_TAG_BITS)`. As I can see, there is a whole bunch of similar cases in ReactOS code...

Reply:
Actually my point was more about the "CLEAR_HANDLE" definition :P